aws/resource_aws_dms_event_subscription.go

```diff
@@ -0,0 +1,251 @@
+		SourceType:       aws.String(d.Get("source_type").(string)),
+	}
+
+	if v, ok := d.GetOk("event_categories"); ok {
+		request.EventCategories = expandStringList(v.(*schema.Set).List())
+	}
+
+	log.Println("[DEBUG] DMS update event subscription:", request)
+
+	_, err := conn.ModifyEventSubscription(request)
+
+	if err != nil {
+		return fmt.Errorf("Error updating DMS event subscription: %s", err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"modifying"},
+		Target:     []string{"active"},
+		Refresh:    resourceAwsDmsEventSubscriptionStateRefreshFunc(conn, d.Id()),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		MinTimeout: 10 * time.Second,
+		Delay:      10 * time.Second, // Wait 30 secs before starting
+	}
+
+	// Wait, catching any errors
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("error waiting for DMS event subscription (%s) modification: %s", d.Id(), err)
+	}
+
+	return resourceAwsDmsEventSubscriptionRead(d, meta)
+}
+
+func resourceAwsDmsEventSubscriptionRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dmsconn
+
+	request := &dms.DescribeEventSubscriptionsInput{
+		SubscriptionName: aws.String(d.Id()),
+	}
+
+	log.Println("[DEBUG] DMS read event subscription:", request)
+
+	response, err := conn.DescribeEventSubscriptions(request)
+
+	if isAWSErr(err, dms.ErrCodeResourceNotFoundFault, "") {
+		log.Printf("[WARN] DMS event subscription (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("Error reading DMS event subscription: %s", err)
+	}
+
+	if response == nil || response.EventSubscriptionsList == nil || len(response.EventSubscriptionsList) == 0 {
+		log.Printf("[WARN] DMS event subscription (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	subscription := response.EventSubscriptionsList[0]
+
+	d.Set("enabled", subscription.Enabled)
+	d.Set("sns_topic_arn", subscription.SnsTopicArn)
+	d.Set("source_type", subscription.SourceType)
+	d.Set("name", d.Id())
+	d.Set("event_categories", flattenStringList(subscription.EventCategoriesList))
+	d.Set("source_ids", flattenStringList(subscription.SourceIdsList))
+
+	return nil
+}
+
+func resourceAwsDmsEventSubscriptionDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dmsconn
+
+	request := &dms.DeleteEventSubscriptionInput{
+		SubscriptionName: aws.String(d.Get("name").(string)),
+	}
+
+	log.Println("[DEBUG] DMS event subscription delete:", request)
+
+	_, err := conn.DeleteEventSubscription(request)
+
+	if err != nil {
+		return fmt.Errorf("Error deleting DMS event subscription: %s", err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"deleting"},
+		Target:     []string{},
+		Refresh:    resourceAwsDmsEventSubscriptionStateRefreshFunc(conn, d.Id()),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
```

Comment:
While in the `Delete` function, this should use the delete timeout:

```suggestion
		Timeout:    d.Timeout(schema.TimeoutDelete),
```